FInal Budget -app

ecekpo opened the pull request:
**Added**
- I made all the project requirements
- https://budget-app-bmmv.onrender.com/
-  video recording link https://drive.google.com/file/d/15xw2ZmqO2e-ciFrK2iCmeE9Mql-xqY6G/view?usp=sharing

ecekpo committed:
add project setup and no linter errors

ecekpo committed:
Merge pull request #1 from ecekpo/milestone1-projectsetup

Milestone1: Project setup

ecekpo committed:
Add scafold for User Group Expense and GroupExpense

ecekpo committed:
Add association to expense.rb file

ecekpo committed:
Add association to group.rb file

ecekpo committed:
Add foreign key to user.rb

ecekpo committed:
Add update create_group and create expense migration file

ecekpo committed:
Add route for user group and expense

ecekpo committed:
Add rspec rails for unit test

ecekpo committed:
Add validation for name and expense

ecekpo committed:
setup validation for group model

ecekpo committed:
setup validation for user model

ecekpo committed:
added files installation for unit testing

ecekpo committed:
add expense rspec test for validation

ecekpo committed:
Add group rspec unit test

ecekpo committed:
Add User Rspec test for validation

ecekpo committed:
add expense rspec test for validation

ecekpo committed:
add expense rspec test for validation

ecekpo committed:
Fix linter errors

ecekpo committed:
Merge pull request #2 from ecekpo/milestone2-model

Milestone2 model, validations and tests

ecekpo committed:
update dev

ecekpo committed:
Fix linters

ecekpo committed:
Update application.css

mukhammadrizooff commented on app/assets/stylesheets/application.css:
```diff
@@ -0,0 +1,96 @@
+@import 'font-awesome';
+
+* {
```
- [ ] [REQUIRED] Good job so far! Please you need to work on your project's design(`Register`, `Home` and `Detail` pages), I compare your and [Behance](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=) designs pictures below🙏🏼


| Your Design  |
| ------------- | 

#
![image](https://user-images.githubusercontent.com/63915024/221377839-5488bf26-9d69-4aac-88df-6fc8bbe2fa8d.png)
#

| [Behance Design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=)  |
| ------------- |  

#
![image](https://user-images.githubusercontent.com/63915024/221377829-38a0a6ba-78a8-42b7-ab01-da59614176c2.png)
#

ecekpo replied:
@mukhammadrizooff ,
I have implemented what I could on the project because I was limited on time.

mukhammadrizooff commented on README.md:
```diff
@@ -0,0 +1,184 @@
+
+## 🚀 Live Demo <a name="live-demo"></a>
+ 
+
+- [Live Demo Link]()
+
```
- [ ] [REQUIRED] Please you should be deploy your application using `Heroku`, `Netlify` or `GitHub Pages` because the project should deployed properly, the `home` and `detail` page works as expected 🙏🏼 

![image](https://user-images.githubusercontent.com/63915024/221377521-ac021695-de0e-4ba6-b0ca-8154ab0e44e1.png)

ecekpo replied:
My project was deployed https://budget-app-bmmv.onrender.com/

mukhammadrizooff replied:
> My project was deployed https://budget-app-bmmv.onrender.com/

@ecekpo Nice, please add this link to your README which required place 👍🏼

mukhammadrizooff commented on README.md:
```diff
@@ -0,0 +1,184 @@
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the [issues page](../../issues/).
+
+## Credit
+   I give credit to the original design idea by Gregoire Vella on Behance.
+
```
- [ ] [REQUIRED] The README contains an "Acknowledgements" that give appropriate credit, However, I will suggest you add the name [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) as the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/). This will give more description of the author.

![221359671-d46e3f22-91a6-4fa1-ae39-e3d39e6f985e](https://user-images.githubusercontent.com/63915024/221377751-689733b4-d895-4c99-b54a-8ac001353320.png)

mukhammadrizooff commented on README.md:
```diff
@@ -0,0 +1,184 @@
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./MIT.md) licensed.
+
+_NOTE: we recommend using the [MIT license](https://choosealicense.com/licenses/mit/) - you can set it up quickly by [using templates available on GitHub](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository). You can also use [any other license](https://choosealicense.com/licenses/) if you wish._
```
- [ ] [REQUIRED] Good job so far! It woud be proffesional if you create `MIT` license for your application after create `MIT` remove the `line 184` 🙏🏼

mukhammadrizooff reviewed (changes requested):
Hi @ecekpo, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

<img src="https://media2.giphy.com/media/Jq824R93JsLwZCaiSL/giphy.gif"/>

## Highlights
- No linter errors ✔️ 
- Correct `GitFlow` ✔️ 
- Good implementations ✔️ 

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ecekpo committed:
Delete LICENSE

ecekpo committed:
update changes

ecekpo committed:
Merge branch 'dev' of https://github.com/ecekpo/Budget-app into dev

ecekpo committed:
Update application.css

ecekpo committed:
Update application.css

ecekpo committed:
Update application.css

ecekpo committed:
include live demo

ecekpo committed:
Merge branch 'dev' of https://github.com/ecekpo/Budget-app into dev

codepantha commented on app/assets/stylesheets/application.css:
```diff
@@ -0,0 +1,136 @@
+@import 'font-awesome';
+
+* {
```
- [ ] I acknowledge that you have worked very hard on this project, but just like the previous reviewer suggested, I recommend that you ensure your design looks like the given design guidelines especially the signup and transactions pages. For the signup page, simply style the form inputs (color, font, layout) as closely as possible to the given design guideline.
## Behance Design
![image](https://user-images.githubusercontent.com/63915024/221377829-38a0a6ba-78a8-42b7-ab01-da59614176c2.png)
## Your Design
![image](https://user-images.githubusercontent.com/63915024/221377839-5488bf26-9d69-4aac-88df-6fc8bbe2fa8d.png)

ecekpo replied:
@codepantha,
I implemented it and it's working from my end

codepantha reviewed (changes requested):
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @ecekpo 👋, 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

<img src="https://media2.giphy.com/media/rZ10JtoPRv1Ly/giphy.gif"/>

## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ No linter errors

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ecekpo commented:
![Screenshot (294)](https://user-images.githubusercontent.com/101729024/221387028-8cbef59a-921c-4e33-b233-fac370625e1e.png)

ecekpo commented:
![screenshot2](https://user-images.githubusercontent.com/101729024/221387250-30ad70a6-c2c6-40b3-977a-49248991ddb7.png)
![screenshot3](https://user-images.githubusercontent.com/101729024/221387253-d28ea3aa-bc52-40d3-b660-e66f801558ef.png)

ecekpo commented:
@codepantha,
Find attached screenshot from my end

Lameck1 reviewed (changes requested):
Hello @ecekpo,

This is a manual review.

I want you to know that you've done well in addressing the concerns that were previously raised. You only forgot to do one thing, which is to redeploy the application, so that the changes you have made in the project may be visible on the deployed live version. This is the reason why the last reviewer still suggested the same changes that were already asked before.

Please redeploy the project and all will be good.

Happy coding, and keep rocking 🚀.

ecekpo committed:
update registration page

ecekpo committed:
Update README.md

ecekpo commented:
> Hello @ecekpo,
> 
> This is a manual review.
> 
> I want you to know that you've done well in addressing the concerns that were previously raised. You only forgot to do one thing, which is to redeploy the application, so that the changes you have made in the project may be visible on the deployed live version. This is the reason why the last reviewer still suggested the same changes that were already asked before.
> 
> Please redeploy the project and all will be good.
> 
> Happy coding, and keep rocking 🚀.

Hello @Lameck1 ,

Thank you so much for the review.

This is the link for the re-deployed Budget project as requested.

The link will be available [here](https://budget-app-bmmv.onrender.com)
